src/main/java/org/cbio/gdcpipeline/reader/MutationReader.java

```diff
@@ -139,36 +147,29 @@ private void addRecord(MutationRecord newRecord, String maf_filename) {
         }
     }
 
-    private boolean identicalVariant(MutationRecord record, MutationRecord newRecord) {
-        return record.getTumor_Sample_Barcode().equals(newRecord.getTumor_Sample_Barcode()) &&
-                record.getChromosome().equals(newRecord.getChromosome()) &&
-                record.getStart_Position().equals(newRecord.getStart_Position()) &&
-                record.getEnd_Position().equals(newRecord.getEnd_Position()) &&
-                record.getStrand().equals(newRecord.getStrand()) &&
-                record.getReference_Allele().equals(newRecord.getReference_Allele()) && (sameTumourSequence(record, newRecord));
-    }
-
-    private boolean sameTumourSequence(MutationRecord record, MutationRecord newRecord) {
-        if (!(record.getTumor_Seq_Allele1().equals(record.getReference_Allele()) || record.getTumor_Seq_Allele1().isEmpty() || CommonDataUtil.hasMissingKeys(record.getTumor_Seq_Allele1()))) {
-            return record.getTumor_Seq_Allele1().equals(newRecord.getTumor_Seq_Allele1());
-        }
-        return record.getTumor_Seq_Allele2().equals(newRecord.getTumor_Seq_Allele2());
-    }
-
     private FieldSetMapper<MutationRecord> mutationFieldSetMapper() {
         return (FieldSetMapper<MutationRecord>) (FieldSet fs) -> {
             MutationRecord record = new MutationRecord();
             for (String header : record.getHeader()) {
                 try {
-                    record.getClass().getMethod("set" + header, String.class).invoke(record, fs.readString(header));
+                    record.getClass().getMethod("set" + header.toUpperCase(), String.class).invoke(record, fs.readString(header));
                 } catch (Exception e) {
                     if (LOG.isDebugEnabled()) {
                         LOG.error(" Error in setting record for :" + header);
                     }
                     e.printStackTrace();
                 }
             }
-            return record;
+            //annotate
+            AnnotatedRecord annotatedRecord = new AnnotatedRecord();
+            try {
+                annotatedRecord = annotator.annotateRecord(record, false, "uniprot", true);
+            } catch (HttpServerErrorException e) {
+                if (LOG.isWarnEnabled()) {
+                    LOG.warn("Failed to annotate a record from json! Sample: " + record.getTUMOR_SAMPLE_BARCODE() + " Variant: " + record.getCHROMOSOME() + ":" + record.getSTART_POSITION() + record.getREFERENCE_ALLELE() + ">" + record.getTUMOR_SEQ_ALLELE2());
+                }
+            }
+            return (MutationRecord)annotatedRecord;
```

Comment:
I don't like this - could lead to issues down the line. Just make the subsequent processor/writer expect an `AnnotatedRecord`.

Reply:
done